sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -453,6 +454,160 @@ class wi_element<uint16_t, NumRows, NumCols, Layout, Group> {
+  friend type operator op(                                                     \
+      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
+                       NumCols, Layout, Group> &lhs,                           \
+      const sycl::ext::oneapi::experimental::bfloat16 &rhs) {                  \
+    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) op rhs;           \
+  }                                                                            \
+  friend type operator op(                                                     \
+      const sycl::ext::oneapi::experimental::bfloat16 &lhs,                    \
+      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
+                       NumCols, Layout, Group> &rhs) {                         \
+    return __spirv_VectorExtractDynamic(rhs.M.spvm, rhs.idx) op lhs;           \
+  }
+  OP(sycl::ext::oneapi::experimental::bfloat16, +)
+  OP(sycl::ext::oneapi::experimental::bfloat16, -)
+  OP(sycl::ext::oneapi::experimental::bfloat16, *)
+  OP(sycl::ext::oneapi::experimental::bfloat16, /)
+#undef OP
+#define OP(type, op)                                                           \
+  friend type operator op(                                                     \
+      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
+                       NumCols, Layout, Group> &lhs,                           \
+      const sycl::ext::oneapi::experimental::bfloat16 &rhs) {                  \
+    return type{static_cast<float>(__spirv_VectorExtractDynamic(               \
+        lhs.M.spvm, lhs.idx)) op static_cast<float>(rhs)};                     \
+  }                                                                            \
+  friend type operator op(                                                     \
+      const sycl::ext::oneapi::experimental::bfloat16 &lhs,                    \
+      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
+                       NumCols, Layout, Group> &rhs) {                         \
+    return type{static_cast<float>(__spirv_VectorExtractDynamic(               \
+        rhs.M.spvm, rhs.idx)) op static_cast<float>(lhs)};                     \
+  }
+  OP(bool, ==)
+  OP(bool, !=)
+  OP(bool, <)
+  OP(bool, >)
+  OP(bool, <=)
+  OP(bool, >=)
+#undef OP
+#else // __SYCL_DEVICE_ONLY__
+#define OP(type, op)                                                           \
+  friend type operator op(                                                     \
+      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
+                       NumCols, Layout, Group> &lhs,                           \
+      const sycl::ext::oneapi::experimental::bfloat16 &rhs) {                  \
+    (void)lhs;                                                                 \
+    (void)rhs;                                                                 \
```

Comment:
Instead just remove the names from the `op()` declaration

Reply:
i didn't get your point. actually this code is for host. the param list of friend type operator op should be the same as line555 which is for device code.

Reply:
I believe what @keryell suggests is to change this to

```
  friend type operator op(                                                     \
      const wi_element<sycl::ext::oneapi::experimental::bfloat16, NumRows,     \
                       NumCols, Layout, Group> &,                              \
      const sycl::ext::oneapi::experimental::bfloat16 &) {                     \
    throw runtime_error("joint matrix is not supported on host device.",       \
                        PI_INVALID_DEVICE);                                    \
  }  
```

which preserves the signature but doesn't have unused arguments as they are unnamed.